pts-core/commands/check_tests.php

```diff
@@ -0,0 +1,535 @@
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/**
+ * check_tests
+ * 
+ * Performs a check against the vendor on the XML elements in the test-profiles downloads.xml file. 
+ * Checks performed include filename, filesize, SHA256 and MD5 hash.
+ * 
+ * Status is output to a PTS_OPENBENCHMARKING_SCRATCH_PATH/checkTestResults.json file.
+ * 
+ */
+class check_tests implements pts_option_interface
+{
+	const doc_section = 'Information';
+	const doc_description = 'This option will perform a check on one or more test profiles to determine if there have been any vendor changes to the filename, filesize, url location, md5 and sha256 checksums.';
+
+	// The file holding the final json results
+	const JSON_FILE = PTS_OPENBENCHMARKING_SCRATCH_PATH . 'check-tests-results.json';
+
+	// File contains a list of all the tests that have already been tested
+	const TESTED_FILES = PTS_OPENBENCHMARKING_SCRATCH_PATH . 'check-tests-tested.txt';
+
+	// Delimiter in the TESTED_FILES
+	const DELIMITER = ":::";
+
+	// Downloaded file from vendors
+	const DOWNLOADED_VENDOR_FILES = PTS_OPENBENCHMARKING_SCRATCH_PATH . "checkTestsDownloads" . "/";
+
+	// Number of child process to spawn
+	const CHILD_PROCS = 4;
+
+	// Keys used in Assocative array to store vendor data
+	const V_IDENTIFIER = 'identifier';			// name and version of the test profile
+	const V_URL = 'url';						// url from which test profile is downloaded
+	const V_DOWNLOAD_FILE = 'downloadFile';		// name of file downloaded. Usuall basename of url
+	const V_STATUS = 'status';					// http status code for the download
+	const V_DOWNLOAD_TIME = "downloadTime";		// complete time to connect to url and download
+	const V_DOWNLOAD_SIZE = "downloadSize";		// size of downloaded file
+	const V_DUPLICATE = "duplicate";			// indicates if we have already downloaded the file as part of an earlier test profile version
+	const V_REDIRECT = "redirectTo";			// If a url is redirected, this is the original url location. V_URL will contain the redirection 
+
+	/**
+	 * Determines if the test profile is valid. If invalid 'Invalid Arguement' Problem reported.
+	 * */
+	public static function argument_checks()
+	{
+		return array(
+			new pts_argument_check('VARIABLE_LENGTH_MAYBE', array('pts_test_profile', 'is_test_profile'), null)
+		);
+	}
+
+	/**
+	 * Extracts all available test profiles that will be checked. The download.xml file is extracted for
+	 * each test and test are performed on the filename, filesize, sha256 and md5 values to ensure it is
+	 * concurrent with the Service Provider (ie john-the-ripper).
+	 * 
+	 * @param $r 	Checks named test(s) only 
+	 */
+	public static function run($r)
+	{
+		if (!function_exists('curl_init')) {
+			echo pts_client::cli_colored_text("Test Failed. cURL must be installed to proceed." . PHP_EOL . PHP_EOL, 'red', true);
```

Comment:
Missing a return false after the error message. Otherwise the code will still proceed and then hit the fatal error.

Reply:
Updated.

Reply:
Done